core/src/main/scala/org/ensime/model/ModelBuilders.scala

```diff
@@ -243,14 +243,38 @@ trait ModelBuilders {
     def fromSymbol(sym: Symbol, relevance: Int): CompletionInfo =
       fromSymbolAndType(sym, sym.tpe, relevance)
 
-    def fromSymbolAndType(sym: Symbol, tpe: Type, relevance: Int): CompletionInfo =
+    def fromSymbolAndType(sym: Symbol, tpe: Type, relevance: Int): CompletionInfo = {
+      val typeInfo = TypeInfo(tpe)
       CompletionInfo(
-        Some(TypeInfo(tpe)),
+        Some(typeInfo),
         sym.nameString,
         relevance,
-        None
+        None,
+        isInfix(sym.nameString, typeInfo)
       )
+    }
+
+    private def isInfix(symName: String, typeInfo: TypeInfo): Boolean = {
+
+      def eligibleForInfix(symName: String, paramSections: Iterable[ParamSectionInfo]) = {
+        val MAX_NAME_LENGTH_FOR_INFIX = 3
+        val PARAMETER_SETS_NUM_FOR_INFIX = 1
+        val PARAMETER_LIST_SIZE_FOR_INFIX = 1
+        val BLACKLIST = Set("map")
```

Comment:
it might be marginally more efficient to put this in an object, so it isn't recreated on each invocation (ints are interned but not heap objects). Convention for these sorts of things so far has been "excluded" or "banned" list because it's less ambiguous. We plan on making this stuff user configurable at some point, see #1346